src/platform/api/pythonApi.ts

```diff
@@ -548,19 +548,31 @@ export class InterpreterService implements IInterpreterService {
                         .join(', ')}`
                 );
                 await Promise.all(
-                    api.environments.known.map(async (item) => {
-                        try {
-                            const env = await api.environments.resolveEnvironment(item.id);
-                            const resolved = this.trackResolvedEnvironment(env);
-                            if (resolved) {
-                                allInterpreters.push(resolved);
-                            } else {
-                                traceError(`Failed to get env details from Python API for ${item.id} without an error`);
+                    api.environments.known
+                        .filter((item) => this.isValidWorkSpaceRelatedEnvironment(item))
+                        .map(async (item) => {
+                            try {
+                                const env = await api.environments.resolveEnvironment(item.id);
+                                const resolved = this.trackResolvedEnvironment(env);
+                                traceVerbose(
+                                    `Python environment ${env?.id} from Python Extension API is ${JSON.stringify(
+                                        env
+                                    )} and translated is ${JSON.stringify(resolved)}`
+                                );
+                                if (!this.isValidWorkSpaceRelatedEnvironment(item)) {
+                                    return;
```

Comment:
Filter after we get the details, in case the previous env information was partial (empty)